server/models/question.js

```diff
@@ -1,12 +1,23 @@
 const mongoose = require('mongoose');
 
-const questionSchema = mongoose.Schema({
-  title: String,
-  question: String,
-  user: { default: 'user', type: String },
-}, {
-  timestamps: true,
-});
+const questionSchema = mongoose.Schema(
+  {
+    submitted_at: Date.now(),
+    title: String,
+    body: String,
+    votes: Number,
+    author: {
+      _id: {
+        $oid: ObjectId(),
```

Comment:
`ObjectId` is not defined and maybe we should add a new `author` schema like in the [docs](http://mongoosejs.com/docs/subdocs.html)?

Reply:
I agree with this. Would make the DB easier to manage.

Reply:
I think we can add in a new schema for the `author` but we still need to keep a reference in the `question` schema. Would a string be better or should we just use an `ObjectId` that we would get from the new schema?

Reply:
The author schema should be no more than a simple `{ name: String, avatar: String }` we don't need to add the `_id` field here, at least I don't think so. But I don't know mongoose :( Could you test it this way?